airflow/providers/google/cloud/triggers/dataproc.py

```diff
@@ -140,24 +153,75 @@ def serialize(self) -> tuple[str, dict[str, Any]]:
                 "gcp_conn_id": self.gcp_conn_id,
                 "impersonation_chain": self.impersonation_chain,
                 "polling_interval_seconds": self.polling_interval_seconds,
+                "delete_on_error": self.delete_on_error,
             },
         )
 
     async def run(self) -> AsyncIterator[TriggerEvent]:
-        while True:
-            cluster = await self.get_async_hook().get_cluster(
-                project_id=self.project_id, region=self.region, cluster_name=self.cluster_name
+        """Run the trigger."""
+        try:
+            while True:
+                cluster = await self.fetch_cluster()
+                state = cluster.status.state
+                if state == ClusterStatus.State.ERROR:
+                    await self.delete_when_error_occurred(cluster)
+                    yield TriggerEvent(
+                        {
+                            "cluster_name": self.cluster_name,
+                            "cluster_state": state.ERROR,
```

Comment:
since `state = ClusterStatus.State.ERROR` after entering the if block, I don't think `state.ERROR` which would translate to `ClusterStatus.State.ERROR.ERROR` or `ClusterStatus.State.ERROR.DELETING` is the right usage. Can you check the type of `state` variable and if `state.DELETING` or `state.ERROR` is available on that type? 

I think it could be, but suggest to check the type and ensure that we've the right usage and right trigger event value yielded in the worker.
```suggestion
                            "cluster_state": ClusterStatus.State.DELETING,
```

Reply:
I think a better idea would just be to return the state as it is as it is being fetched from the cluster

Reply:
But that would be an inaccurate state no? Since we triggered delete action after that state? 
Might be good to get the latest state or use  ClusterStatus.State.DELETING

Reply:
ClusterStatus.State.DELETING is better